pkg/distributor/distributorerror/error.go

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package distributorerror
+
+import "github.com/grafana/mimir/pkg/util/validation"
+
+type Error interface {
+	error
+	DistributorError()
+}
+
+// DistributorPushError is a generic error returned on
+// distributor's write path.
+type DistributorPushError struct {
+	err error
+}
+
+// Error makes DistributorPushError implement error interface.
+func (e DistributorPushError) Error() string {
+	return e.err.Error()
+}
+
+// Unwrap makes DistributorPushError implement Wrapper interface.
+func (e DistributorPushError) Unwrap() error {
+	return e.err
+}
+
+// PushError makes DistributorPushError implement pusherror.Error interface.
+func (e DistributorPushError) PushError() {}
+
+// Error makes DistributorPushError implement Error interface.
+func (e DistributorPushError) DistributorError() {}
+
+// NewDistributorPushError wraps the given error into a DistributorPushError.
+func NewDistributorPushError(err error) DistributorPushError {
+	return DistributorPushError{
+		err: err,
+	}
+}
+
+// ReplicasNotMatchDistributorPushError is an implementation of Error,
+// corresponding to distributor.replicasNotMatchError.
+type ReplicasNotMatchDistributorPushError struct {
+	DistributorPushError
+}
+
+func NewReplicasNotMatchDistributorPushError(err error) ReplicasNotMatchDistributorPushError {
```

Comment:
This is duplicate of `replicasNotMatchError`, we don't need another one.

Reply:
Now that the package has been moved under `distributor/distributorerror` it makes sense. I can move `tooManyClustersError ` and `replicasNotMatchError` under `distributorerror`.